File: leetcode/arrays/deleting/RemoveDuplicatesInNonSortedArray.java
package arrays.deleting;

import java.util.HashSet;
import java.util.Set;

public class RemoveDuplicatesInNonSortedArray {
    public static void main(String[] args) {
        int[] nums = new int[]{3,4,3,4,5,1,2,3,4,5}; 
        RemoveDuplicatesInNonSortedArray.execute(nums);
        //Should be {1,2}
    }
    
    public static void execute(int[] nums) {
        int i = 0;
        Set<Integer> alreadyProcessed = new HashSet<>();
        for (int j = 0; j < nums.length; j++) {
            if (!alreadyProcessed.contains(nums[j])) {
                nums[i] = nums[j];
                i++;
            }
        }

        for (int j = i; j < nums.length; j++) {
            nums[j] = 0;
        }
    }
}

File: leetcode/arrays/deleting/RemoveDuplicatesInNonSortedArray.py
# [3,4,3,4,5,1,2,3,4,5]
# expected [3,4,5,1,2], k=5

def RemoveDuplicatesInNonSortedArray(nums: list[int]) -> int:
    already_processed = set()
    i: int = 0
    
    for j in range(len(nums)):
        if nums[j] not in already_processed:
            nums[i] = nums[j]
            i += 1
            already_processed.add(nums[j])
    
    for j in range(i, len(nums)):
        nums[j] = 0
        
    return i+1

print(RemoveDuplicatesInNonSortedArray([3,4,3,4,5,1,2,3,4,5]))
print(RemoveDuplicatesInNonSortedArray([3,4,3]))